src/sent.rs

```diff
@@ -153,10 +169,49 @@ impl SentPackets {
         self.congestion_ctrl.on_transmit(seq_num, transmit).unwrap();
     }
 
+    /// Handle an ACK for a packet with sequence number `ack_num`, and an optional selective_ack.
+    ///
+    /// Returns Error, if `ack_num` is not within the sequence number range of the sent packets.
+    pub fn on_ack(
+        &mut self,
+        ack_num: u16,
+        selective_ack: Option<&SelectiveAck>,
+        delay: Duration,
+        now: Instant,
+    ) -> Result<(CircularRangeInclusive, Vec<u16>), SentPacketsError> {
+        let range = self.seq_num_range();
+        if range.contains(ack_num) {
```

Comment:
IMO cleaner to short-circuit for error and then remaining code does not need to be indented

```suggestion
        if !range.contains(ack_num) {
            return Err(SentPacketsError::InvalidAckNum);
        }
        // remaining code here...
```